opentelekomcloud/resource_opentelekomcloud_images_ims_v2.go

```diff
@@ -0,0 +1,293 @@
+			Delete: schema.DefaultTimeout(3 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"image_tags": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: false,
+			},
+			"max_ram": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"min_ram": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: false,
+			},
+			// instance_id is required for creating an image from an ECS
+			"instance_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"image_url"},
+			},
+			// image_url and min_disk are required for creating an image from an OBS
+			"image_url": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"instance_id"},
+			},
+			"min_disk": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				ForceNew:      false,
+				ConflictsWith: []string{"instance_id"},
+			},
+			// following are valid for creating an image from an OBS
+			"os_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"is_config": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  false,
+			},
+			"cmk_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"ECS", "FusionCompute", "BMS", "Ironic",
+				}, true),
+			},
+			// following are additional attributus
+			"visibility": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"disk_format": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceContainerImageTags(d *schema.ResourceData) []cloudimages.ImageTag {
+	var tags []cloudimages.ImageTag
+
+	image_tags := d.Get("image_tags").(map[string]interface{})
+	for key, val := range image_tags {
+		tagRequest := cloudimages.ImageTag{
+			Key:   key,
+			Value: val.(string),
+		}
+		tags = append(tags, tagRequest)
+	}
+	return tags
+}
+
+func getImagetagsToList(d *schema.ResourceData) []string {
+	var tags []string
+
+	image_tags := d.Get("image_tags").(map[string]interface{})
+	for key, val := range image_tags {
+		// key.value
+		tagRequest := key + "." + val.(string)
+		tags = append(tags, tagRequest)
+	}
+	return tags
+}
+
+func resourceImagesImsV2Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ims_Client, err := config.imageV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating OpenTelekomCloud EVS storage client: %s", err)
```

Comment:
should be IMS client here.

Reply:
@niuzhenguo Thanks for your comments, please review it in the new commits.